formatencoder.go

```diff
@@ -117,20 +117,18 @@ func (se TransformEncoder) EncodeEntry(ent zapcore.Entry, fields []zapcore.Field
 		return buf, err
 	}
 	repl.Map(func(key string) (interface{}, bool) {
-		path := strings.Split(key, ">")
-		value, dataType, _, err := jsonparser.Get(buf.Bytes(), path...)
-		if err != nil {
-			return nil, false
-		}
-		switch dataType {
-		case jsonparser.NotExist:
-			return nil, false
-		case jsonparser.Array, jsonparser.Boolean, jsonparser.Null, jsonparser.Number, jsonparser.Object, jsonparser.String, jsonparser.Unknown:
-			// if a value exists, return it as is. A byte is a byte is a byte. The replacer handles them just fine.
-			return value, true
-		default:
+		if strings.Contains(key, ":-") {
+			for _, slice := range strings.Split(key, ":-") {
+				val, noerr := getValue(buf, slice)
```

Comment:
How about `found` instead of `noerr` here? It took me a bit to re-orient my head that we're checking for presence, and not an error. It's also more idiomatic.

Reply:
@mohammed90 

Hi, 

Thanks for taking the time to review the changes, i updated the variable name, however im struggling to define the feature name. Should it be called "Alternative", "defualt" or `fallback` value

Reply:
IMO, "default value". We have precedent in https://caddyserver.com/docs/caddyfile/concepts#environment-variables, which simply uses `:`.

Reply:
> IMO, "default value". We have precedent in https://caddyserver.com/docs/caddyfile/concepts#environment-variables, which simply uses `:`.

Do you think i should just use `:` ? im not too attached to any of them, i simply used `:-` as it's popular with bash scripting

Reply:
I have no horse in this race. When I implemented default value support for Caddyfile env vars, I used `:-` initially as well and @mholt decided he wanted `:`. But I don't maintain this repo, so IMO it's @mohammed90's call.

Reply:
I think being consist is better, I'll do the changes and let @mohammed90 make the call.

Reply:
Good call, Francis! We went with `>` for consistency, so let's also go with `:` here as well. Sorry, @ArabCoders :) I agree with you on the value being an `alternative` though. A default value gives a sense of being static, always being returned and known in advance.

Reply:
Done. Please re-review the changes. And sorry for taking your times.